front/src/common/authorization/components/GrantsManagerSubjects.tsx

```diff
@@ -0,0 +1,140 @@
+          <>
+            <div className="grant-manager-subject-search-section">
+              <span className="grant-manager-subject-search-combobox">
+                <ComboBox
+                  id="add-user-combobox"
+                  value={selectedUser}
+                  suggestions={searchResults}
+                  getSuggestionLabel={(option) => option.name}
+                  onChange={(e) => setSearchTerm(e.target.value)}
+                  onSelectSuggestion={(suggestion) => setSelectedUser(suggestion)}
+                  resetSuggestions={() => {}}
+                  autoComplete="off"
+                  narrow
+                />
+              </span>
+              <span className="grant-manager-subject-select">
+                <Select
+                  id="add-user-grant-selector"
+                  getOptionLabel={(option) => option.label}
+                  getOptionValue={(option) => option.value}
+                  onChange={(option) => {
+                    setSelectedUser((prev) =>
+                      prev
+                        ? {
+                            ...prev,
+                            grant: option?.value as Grant,
+                          }
+                        : undefined
+                    );
+                  }}
+                  {...generateGrantSelectProps({
+                    resourceGrants,
+                    subjectGrant: selectedUser?.grant ?? DEFAULT_GRANT,
+                    connectedUserGrant: userGrant,
+                    t,
+                  })}
+                  narrow
+                />
+              </span>
+            </div>
+            <button
+              type="button"
+              className="grant-manager-subject-search-add-button"
+              onClick={() => {
+                if (selectedUser) {
+                  updateUserResourceGrant(selectedUser.id, selectedUser.grant);
+                  resetSuggestions();
+                }
+                setDisplayUserSearchSection(false);
+              }}
+            >
+              {t('actions.add')}
+            </button>
+          </>
+        )}
+      </div>
+
+      <div className="grant-manager-subject-list">
+        {userSubjectsList?.map(({ id, name, grant }, index) => (
+          <div className="grant-manager-subject-card" key={id}>
+            <span className={cx('grant-manager-subject-name', { bold: index === 0 })}>{name}</span>
+            <span className="grant-manager-subject-select">
+              <Select
+                id={`${id}-${name}`}
+                getOptionLabel={(option) => option.label}
+                getOptionValue={(option) => option.value}
+                onChange={(option) => updateUserResourceGrant(id, option?.value)}
+                {...generateGrantSelectProps({
+                  resourceGrants,
+                  subjectGrant: grant,
+                  connectedUserGrant: userGrant,
+                  t,
+                })}
```

Comment:
Actually, the only params which change here is the subjectGrant. You can simplify the code by adding a new function `generateSelectProps`:
```
const generateSelectProps = (userGrant: Grant | undefined) =>
    generateGrantSelectProps(
      resourceGrants: privilegesByGrant,
      subjectGrant: userGrant ?? DEFAULT_GRANT,
      connectedUserGrant,
      t
    );
```

and then use it in the jsx:
```suggestion
                {...generateSelectProps(grant)}
```